### PR TITLE
Update the swarm notebook example

### PR DESCRIPTION
**Pull request by yiranwu0**
## Checks

- [ ] I've included any doc changes needed for https://ag2ai.github.io/ag2/. See https://ag2ai.github.io/ag2/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb [hunk 1]
```diff
@@ -0,0 +1,694 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Implement Swarm with AutoGen GroupChat\n",
```
Comment (qingyun-wu) [inline notebook/agentchat_swarm_legacy_w_groupchat.ipynb:7]: Keep the doc consistent

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb [hunk 2]
```diff
@@ -0,0 +1,694 @@
...
+    "\n",
+    "\n",
+    "AutoGen offers conversable agents powered by LLM, tool or human, which can be used to perform tasks collectively via automated chat. Recently, OpenAI has released a [Swarm](https://github.com/openai/swarm) framework that focuses on making agent coordination and execution lightweight. In autogen, the groupchat allows customized speaker selection, which can be used to achieve the same orchestration pattern. This feature is also supported by our research paper [StateFlow: Enhancing LLM Task-Solving through State-Driven Workflows](https://autogen-ai.github.io/autogen/blog/2024/02/29/StateFlow).\n",
```
Comment (qingyun-wu) [inline notebook/agentchat_swarm_legacy_w_groupchat.ipynb:10]:
```suggestion
    "AG2 offers conversable agents powered by LLM, tool or human, which can be used to perform tasks collectively via automated chat. Recently, OpenAI has released a [Swarm](https://github.com/openai/swarm) framework that focuses on making agent coordination and execution lightweight. In autogen, the groupchat allows customized speaker selection, which can be used to achieve the same orchestration pattern. This feature is also supported by our research paper [StateFlow: Enhancing LLM Task-Solving through State-Driven Workflows](https://autogen-ai.github.io/autogen/blog/2024/02/29/StateFlow).\n",
```

**Comment by marklysze** [comment]
Thanks for the comments, I've resolved and updated as well. Will also update the blog post, so it can be reviewed again and merged after that.

File: notebook/agentchat_swarm.ipynb [hunk 3]
```diff
@@ -4,12 +4,14 @@
...
+    "AG2 offers conversable agents, powered by LLMs, tools or a human, that can perform tasks collectively via an automated chat. Recently, OpenAI released a [Swarm](https://github.com/openai/swarm) framework that focuses on making agent coordination and execution lightweight. \n",
     "\n",
-    "In this notebook, we implement the [airline customer service example](https://github.com/openai/swarm/tree/main/examples/airline) from OpenAI Swarm.\n"
+    "Now, AG2 offers a simple programming interface to build and orchestrate a swarm of agents. Please checkout out the [Documentation](https://ag2ai.github.io/ag2/docs/topics/swarm) and [Blog](https://ag2ai.github.io/ag2/blog/2024/11/17/Swarm) for more details.\n",
```
Comment (sonichi) [inline notebook/agentchat_swarm.ipynb:12]:
```suggestion
    "Now, AG2 offers a simple programming interface to build and orchestrate a swarm of agents. Please check the [Documentation](https://ag2ai.github.io/ag2/docs/topics/swarm) and [Blog](https://ag2ai.github.io/ag2/blog/2024/11/17/Swarm) for more details.\n",
```

File: notebook/agentchat_swarm.ipynb [hunk 4]
```diff
@@ -203,228 +241,82 @@
...
-    ")\n",
-    "manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=llm_config)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "## Run the code\n",
+    "For this example, it requires human interaction with the agents. So we pass in an `user` agent, and set the `after_work` parameter to `AfterWorkOption.REVERT_TO_USER`. This means, each time a swarm agent finishes its work, the conversation will be reverted back to the user to give the next input.\n",
```
Comment (sonichi) [inline notebook/agentchat_swarm.ipynb:312]:
```suggestion
    "For this example, it requires human interaction with the agents. So we pass in a `user` agent, and set the `after_work` parameter to `AfterWorkOption.REVERT_TO_USER`. This means, each time a swarm agent finishes its work, the conversation will be reverted back to the user to give the next input.\n",
```

File: notebook/agentchat_swarm.ipynb [hunk 5]
```diff
@@ -203,228 +241,82 @@
...
-    "# Human\n",
-    "user = UserProxyAgent(\n",
-    "    name=\"User\",\n",
-    "    system_message=\"Human user\",\n",
-    "    code_execution_config=False,\n",
-    ")"
+    "transfer_to_triage_description = \"Call this function when a user needs to be transferred to a different agent and a different policy.\\nFor instance, if a user is asking about a topic that is not handled by the current agent, call this function.\"\n",
+    "flight_cancel.register_hand_off(ON_CONDITION(triage_agent, transfer_to_triage_description))\n",
+    "flight_change.register_hand_off(ON_CONDITION(triage_agent, transfer_to_triage_description))\n",
+    "lost_baggage.register_hand_off(ON_CONDITION(triage_agent, transfer_to_triage_description))"
```
Comment (sonichi) [inline notebook/agentchat_swarm.ipynb:301]: Is it better to use a for loop?

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb [hunk 6]
```diff
@@ -0,0 +1,709 @@
...
+   "source": [
+    "````{=mdx}\n",
+    ":::tip\n",
+    "This notebook shows you how to implement a swarm-like orchestration using AG2's group chat.\n",
+    "\n",
+    "If you want to get up and running with a swarm faster, we have implemented swarm within AG2, see [this notebook](/docs/notebooks/agentchat_swarm).\n",
+    ":::\n",
+    "````"
+   ]
+  },
```
Comment (sonichi) [inline notebook/agentchat_swarm_legacy_w_groupchat.ipynb:29]: Move this cell to top.

**Comment by sonichi** [comment]
Is there an obvious benefit of the new swarm programming interface vs. the old one? If so, can we talk about them in doc/blogpost?

File: notebook/agentchat_swarm.ipynb (hunk 5 above)
Comment (yiranwu0) [reply notebook/agentchat_swarm.ipynb:301]: Yes, Resolved

**Comment by yiranwu0** [comment]
> Is there an obvious benefit of the new swarm programming interface vs. the old one? If so, can we talk about them in doc/blogpost?

Yes, will update that

**Review by sonichi — APPROVED** [review]
Shall we merge?

File: notebook/agentchat_swarm.ipynb
Comment (sonichi) [inline notebook/agentchat_swarm.ipynb:1]: "description": "Implement Swarm with AutoGen GroupChat",
Can be updated into:
Swarm-style orchestration

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb [hunk 7]
```diff
@@ -0,0 +1,709 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "front_matter": {
+   "description": "Implement Swarm with AutoGen GroupChat",
```
Comment (sonichi) [inline notebook/agentchat_swarm_legacy_w_groupchat.ipynb]:
```suggestion
   "description": "(Legacy) Implement Swarm-style orchestration with GroupChat",
```

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb [hunk 8]
```diff
@@ -0,0 +1,709 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Implement Swarm with AG2's GroupChat\n",
+    "\n",
+    "\n",
+    "AG2 offers conversable agents, powered by LLMs, tools or a human, that can perform tasks collectively via an automated chat. Recently, OpenAI has released a [Swarm](https://github.com/openai/swarm) framework that focuses on making agent coordination and execution lightweight.\n",
+    "\n",
+    "In AutoGen, the groupchat allows customized speaker selection, which can be used to achieve the same orchestration pattern. This feature is also supported by our research paper [StateFlow: Enhancing LLM Task-Solving through State-Driven Workflows](https://ag2ai.github.io/ag2/blog/2024/02/29/StateFlow).\n",
```
Comment (marklysze) [inline notebook/agentchat_swarm_legacy_w_groupchat.ipynb]:
```suggestion
    "In AG2, the groupchat allows customized speaker selection, which can be used to achieve the same orchestration pattern. This feature is also supported by our research paper [StateFlow: Enhancing LLM Task-Solving through State-Driven Workflows](https://ag2ai.github.io/ag2/blog/2024/02/29/StateFlow).\n",
```

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb [hunk 9]
```diff
@@ -0,0 +1,709 @@
...
+    "    system_message=STARTER_PROMPT + LOST_BAGGAGE_POLICY,\n",
+    "    llm_config=llm_config,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> With AutoGen, you don't need to write schemas for functions. You can add decorators to the functions to register a function schema to an LLM-based agent, where the schema is automatically generated.\n",
```
Comment (marklysze) [inline notebook/agentchat_swarm_legacy_w_groupchat.ipynb]:
```suggestion
    "> With AG2, you don't need to write schemas for functions. You can add decorators to the functions to register a function schema to an LLM-based agent, where the schema is automatically generated.\n",
```

**Review by marklysze — CHANGES_REQUESTED** [review]

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb (hunk 8 above)
Comment (marklysze) [reply notebook/agentchat_swarm_legacy_w_groupchat.ipynb]: Sorry, can we update these

Comment (yiranwu0) [reply notebook/agentchat_swarm_legacy_w_groupchat.ipynb]: edited.

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb (hunk 9 above)
Comment (yiranwu0) [reply notebook/agentchat_swarm_legacy_w_groupchat.ipynb]: edited.

File: notebook/agentchat_swarm_legacy_w_groupchat.ipynb (hunk 7 above)
Comment (yiranwu0) [reply notebook/agentchat_swarm_legacy_w_groupchat.ipynb]: edited

**Pull request merged**